src/utils/openShiftConnectorUtils.ts

```diff
@@ -0,0 +1,150 @@
+import { exec, ExecException } from 'child_process';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { commands, Disposable, Extension, extensions, ProgressLocation, Uri, window } from "vscode";
+
+export const OPENSHIFT_CONNECTOR_EXTENSION_ID: string = 'redhat.vscode-openshift-connector';
+export const OPENSHIFT_CONNECTOR = 'OpenShift Connector extension';
+const DOWNLOAD_TIMEOUT: number = 60000; // Timeout for downloading VSCode OpenShift Connector, in milliseconds
+
+/**
+ * Returns true if the OpenShift connector extension is installed, and false otherwise
+ *
+ * @returns true if the OpenShift connector extension is installed, and false otherwise
+ */
+export function isOpenShiftConnectorInstalled(): boolean {
+  return !!extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID);
+}
+
+/**
+ * Returns the OpenShift Connector extension API
+ *
+ * @throws Error if the extension is not installed
+ * @returns the OpenShift Connector extension API
+ */
+export async function getOpenShiftConnector(): Promise<any> {
+  if (!isOpenShiftConnectorInstalled()) {
+    throw new Error(`${OPENSHIFT_CONNECTOR} is not installed`);
+  }
+  const openShiftConnector: Extension<any> = extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID);
+  if (openShiftConnector.isActive) {
+    return openShiftConnector.exports;
+  }
+  return extensions.getExtension(OPENSHIFT_CONNECTOR_EXTENSION_ID).activate();
+}
+
+/**
+ * Install the OpenShift Connector extension
+ *
+ * @returns when the extension is installed
+ * @throws if the user refuses to install the extension, or if the extension does not get installed within a timeout period
+ */
+async function installOpenShiftConnector(): Promise<void> {
```

Comment:
Even better version for this function.
Notice: 
1. no Promise.race() required anymore
2. no unused parameters in callback
3. Promise.resolve(commands.executeCommand()) call converts `thenable` instance to `Promise` and that in turn allows to use just `.catch(reject)`.
4. Even when timer expired after Promise fulfilled, rejecting it won't have any effect on fulfilled promise
5. ?. is not required because it `installListenerDisposable` is always going to be initialized
```
async function installOpenShiftConnector(): Promise<void> {
  let installListenerDisposable: Disposable;

  return new Promise<void>((resolve, reject) => {
    installListenerDisposable = extensions.onDidChange(() => {
      if (isOpenShiftConnectorInstalled()) {
        resolve();
      }
    });
    Promise.resolve(commands.executeCommand("workbench.extensions.installExtension", OPENSHIFT_CONNECTOR_EXTENSION_ID)).catch(reject);
    setTimeout(reject, DOWNLOAD_TIMEOUT, new Error(`${OPENSHIFT_CONNECTOR} installation is taking a while. Cancelling 'Deploy to OpenShift'. Please retry after the OpenShift Connector installation has finished`));
  }).finally(() => {
    installListenerDisposable.dispose();
  });
}
```